lib/project.ts

```diff
@@ -272,12 +273,57 @@ export class Project implements Project.IProject {
 		}).future<string[]>()();
 	}
 
+	public onWPSdkVersionChanging(newVersion: string): IFuture<void> {
+		return ((): void => {
+			if(newVersion === this.projectData["WPSdk"]) {
+				return;
+			}
+
+			var validWPSdks = this.getSupportedWPFrameworks().wait();
+			if(!_.contains(validWPSdks, newVersion)) {
+				this.$errors.fail({ formatStr: util.format("The selected version %s is not supported. Supported versions are %s", newVersion, validWPSdks.join(", ")), suppressCommandHelp: true });
+			}
+
+			this.$logger.info("Migrating to WPSdk version %s", newVersion);
+			if(helpers.versionCompare(newVersion, "8.0") > 0) {
+				// at least Cordova 3.7 is required
+				if(helpers.versionCompare(this.projectData.FrameworkVersion, "3.7.0") < 0) {
+					var cordovaVersions = this.$cordovaMigrationService.getSupportedFrameworks().wait();
+					var currentFramework = _.find(cordovaVersions, cv => cv.Version === this.projectData.FrameworkVersion);
+
+					// Find last framework which is not experimental.
+					var selectedFramework = _.findLast(cordovaVersions, cv => cv.DisplayName.indexOf("Exprimental") == -1);
+					if(helpers.versionCompare(selectedFramework.Version, "3.7.0") < 0) {
+						// if latest stable framework version is below 3.7.0, find last 'Experimental'.
+						selectedFramework = _.findLast(cordovaVersions, cv => helpers.versionCompare("3.7.0", cv.Version) < 0);
+					}
+
+					var shouldUpdateFramework = this.$prompter.confirm(util.format("You are trying to use version of Windows Phone SDK that is not supported for your project's Cordova version. Do you want to use %s?", selectedFramework.DisplayName)).wait()
+					if(shouldUpdateFramework) {
+						this.onFrameworkVersionChanging(selectedFramework.Version).wait();
+					} else {
+						this.$errors.fail({ formatStr: util.format("Unable to use Windows Phone SDK %s as the current Cordova version %s does not support it. You should target at least Cordova 3.7.0 in order to use Windows Phone 8.1 SDK.", newVersion, selectedFramework.Version), suppressCommandHelp: true });
```

Comment:
I believe this is the perfect PR to introduce failWithoutHelp (function name pending discussion)